### PR TITLE
feat: optional sql server change tables and transaction logs

### PR DESCRIPTION
**Pull request by nbugden**
Adds optional blocks for `transaction_logs` and `change_tables` to the `sql_server_source_config`.
Fixes https://github.com/hashicorp/terraform-provider-google/issues/18836

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
datastream: added `transaction_logs` and `change_tables` to the `datastream_stream` resource
```

### CODE REVIEW

**Comment by nbugden** [comment]
@BBBmau checking in to see if you can review this PR

File: mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb
Comment (BBBmau) [inline mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb:86]: we'll want to also include a test that uses the `transcation_logs` field also.

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: I refactored this test to be for the `transaction_logs` (as that is the default) and added a test for the `change_tables` in  this commit: a06af09fb62ca6f4d5a8749485dc5d65e390d95f

I set `skip_test: true` based on a [previous conversation](https://github.com/GoogleCloudPlatform/magic-modules/pull/10553#issuecomment-2091882863) in another one of my datastream PRs. To run the tests would require configuration on the SQL Server instance. I noted this in the `Stream.yaml` as well.

Comment (BBBmau) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: tests are looking good, I'll run the test on my machine before approving!

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: Let me know if you have a chance to run the tests. I can provide the debug logs from my local if it helps.

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: Any updates @BBBmau

Comment (sergeyyegournov) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: Waiting for this work by @nbugden PR to be merged as well @BBBmau

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: @BBBmau any updates? Would like to wrap this up.

Comment (BBBmau) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: @nbugden  apologies for the delay, only thing left for me is to manual test this. I'll be sure to do this before the end of the week.

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: If it helps I can provide the debug logs from my local. Manual testing will require you configure CDC on the table for testing the `change_tables`

Comment (BBBmau) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: This would be helpful, providing a simple tfconfig for reproduction would also help as well!

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: Adding @yuval-badal to the convo as he provided the DBA expertise for sqlserver on my end.

@BBBmau Attached is:
* example.tf - with example config for setting up the stream
* change_tables.output - tf debug logs when creating a stream using the new `change_tables` block
* transaction_logs.output - tf debug logs when updating the stream using the new `transaction_logs` block
* original.output - tf debug logs when updating the stream by removing the `transactio_logs` block

A few things to note:
* I didn't include config for the sql server instance. This is out of scope what the PR since SQL server config would be the repsonsiblity of the DBA. That said, for this to work it requires you have a SQL Server instance set up with:
  * CDC enabled on the db and the table
    Enable CDC on the DB
    ```sql
    USE db;
    EXEC sys.sp_cdc_enable_db;
    ```
    Enable CDC on the Table
    ```sql
    EXEC sys.sp_cdc_enable_table
    @source_schema = N'test',
    @source_name = N'test_table',
    @supports_net_changes = 0,
    @role_name = NULL; 
    ```
  * By default you can set up streams for change_tables. To test the transaction_logs you must update the polling interval.
    ```sql
    EXEC sys.sp_cdc_change_job @job_type = 'capture' , @pollinginterval = 86399
    EXEC sp_cdc_stop_job 'capture'. 
    EXEC sp_cdc_start_job 'capture'    
    ```

[example.tf.txt](https://github.com/user-attachments/files/16599763/example.tf.txt)
[change_tables.output.txt](https://github.com/user-attachments/files/16600312/change_tables.output.txt)
[transaction_logs.output.txt](https://github.com/user-attachments/files/16600422/transaction_logs.output.txt)
[original.output.txt](https://github.com/user-attachments/files/16600427/original.output.txt)

**Review by sergeyyegournov — APPROVED** [review]

File: mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb
Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: @BBBmau have you had a chance to review?

Comment (nbugden) [reply mmv1/templates/terraform/examples/datastream_stream_sql_server.tf.erb]: @BBBmau any updates?

**Comment by nbugden** [comment]
@BBBmau any updates here? Want to be able to close this out.

**Comment by nbugden** [comment]
@BBBmau can we get an eta on your review?

**Comment by BBBmau** [comment]
> @BBBmau can we get an eta on your review?

apologies for the delay, I'll be taking a look at this today.

File: mmv1/products/datastream/Stream.yaml
Comment (BBBmau) [inline mmv1/products/datastream/Stream.yaml:1026]: for the most part this all looks good. my main concern is the use of `exactly_one_of` since this would mark both fields as required but from one. I believe we would want to use conflicts instead since the resource itself can be used without setting either `transactionLogs` or `changeTables`.

[Related Documentation on `exactly_one_of`](https://googlecloudplatform.github.io/magic-modules/develop/field-reference/#exactly_one_of)

**Comment by BBBmau** [comment]
It's worth mentioning also that due to the upcoming 6.0.0 release, I am able to approve but won't be merging until after the major release is done.

File: mmv1/products/datastream/Stream.yaml
Comment (nbugden) [reply mmv1/products/datastream/Stream.yaml]: Good call. Switched it up c24b5028c430f371a991dfca4fe308947f264839

**Comment by nbugden** [comment]
> It's worth mentioning also that due to the upcoming 6.0.0 release, I am able to approve but won't be merging until after the major release is done.

Thanks for calling this out. When os the 6.0.0 release going out?

**Comment by BBBmau** [comment]
> > It's worth mentioning also that due to the upcoming 6.0.0 release, I am able to approve but won't be merging until after the major release is done.
> 
> Thanks for calling this out. When os the 6.0.0 release going out?

6.0.0 is set to be released next monday August 26th

**Comment by nbugden** [comment]
> > > It's worth mentioning also that due to the upcoming 6.0.0 release, I am able to approve but won't be merging until after the major release is done.
> > 
> > 
> > Thanks for calling this out. When os the 6.0.0 release going out?
> 
> 6.0.0 is set to be released next monday August 26th

Thanks! I made the change to `conflicts` from `exactly_one_of`. Can you review in the meantime?
https://github.com/GoogleCloudPlatform/magic-modules/commit/c24b5028c430f371a991dfca4fe308947f264839

**Review by BBBmau — APPROVED** [review]
apologies for the late review, everything looks good. thanks for including your debug logs!

**Pull request merged**
